tests/generics/tgenerics_various.nim

```diff
@@ -128,7 +128,7 @@ block trefs:
 
 
 block tsharedcases:
-  proc typeNameLen(x: typedesc): int {.compileTime.} =
+  proc typeNameLen(x: typedesc): int =
     result = x.name.len
```

Comment:
Unrelated bug here (seemingly a simpler case of #19916, #20514, #20937), can also be triggered with

```nim
template typeNameLen(x: typedesc): int =
  (static x.name.len)
```